bundles/AdminBundle/public/js/pimcore/asset/tree.js

```diff
@@ -295,31 +295,46 @@
                      };
  
                      if (res.exists) {
-                         var messageBox = new Ext.window.MessageBox();
+                         let applyToAllCheckbox = Ext.create('Ext.form.field.Checkbox', {
+                             boxLabel: t('asset_upload_apply_to_all')
+                         });
+                         let messageBox = new Ext.window.MessageBox({
+                             layout: {
+                                 type: 'vbox',
+                                 align: 'center'
+                             }
+                         });
                          overwriteConfirmMessageBoxes.push(messageBox);
                          messageBox.show({
                              title: t('file_exists'),
                              msg: t('asset_upload_want_to_overwrite').replace('%s', file.name),
                              buttons: Ext.Msg.OK & Ext.Msg.YES & Ext.Msg.NO,
-                             buttonText: { ok: t('asset_upload_overwrite'), yes: t('asset_upload_keep_both'), no: t('asset_upload_overwrite_all') },
+                             buttonText: {
+                                 ok: t('asset_upload_overwrite'),
+                                 yes: t('asset_upload_keep_both'),
+                                 no: t('asset_upload_skip')
+                             },
                              prompt: false,
                              icon: Ext.MessageBox.QUESTION,
                              fn: function (action) {
-                                 if (action === 'ok') {
-                                     uploadFunction(true);
-                                 } else if (action === 'yes') {
-                                     uploadFunction(false);
-                                 } else if (action === 'no') {
-                                     Ext.each(overwriteConfirmMessageBoxes, function(messageBox) {
-                                         if (messageBox) {
-                                             messageBox.down('button[itemId=ok]').fireHandler();
-                                         }
-                                     });
-                                 } else if (action === 'cancel') {
+                                 if (action === 'ok' || action === 'yes') {
```

Comment:
@BlackbitDevs thanks for the improvement.

Thers one small thing i found - the "apply to all" checkbox doesnt work if you choose "skip".

I would suggest to move the IF check a few lines down to only wrap the uploadFunction.
That should do the job.

Reply:
You are right, I applied your suggested solution.